xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/XWikiWarningMessageMacro.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.internal.macro.message;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.localization.ContextualLocalizationManager;
+
+/**
+ * Displays a warning message.
+ *
+ * @version $Id$
+ * @since 16.7.0RC1
+ */
+@Component
+@Named("warning")
+@Singleton
+public class XWikiWarningMessageMacro extends WarningMessageMacro
+{
+    private static final String ICON_PRETTY_NAME_KEY = "rendering.macro.message.icon.alternative.warning";
```

Comment:
So you decided to make this pretty name related to the macro and not to the icon chosen for the macro, right ?

Reply:
We decided in https://forum.xwiki.org/t/accessibility-html-codestyle-icons-along-text-alternatives/13378 that icons should always be used next to a text alternative. This is why I pushed for having the pretty names independent of the icons and to not provide them by the icon manager.

Reply:
Sure, I just wanted to be sure it was the intended goal given the handling of the translations.

Reply:
Yeah, because the alternative of the icon is context dependant. What is used as a `success` icon here could also be a `done` icon in another context.